File: c++/bubblesort.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    vector<int> arr = {5, 2, 9, 1, 5, 6};
    int n = arr.size();

    for (int i = 0; i < n-1; i++) {
        for (int j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                swap(arr[j], arr[j+1]);
            }
        }
    }

    cout << "Sorted array: ";
    for (int x : arr) cout << x << " ";
    cout << endl;
    return 0;
}

File: rust/bubble_sort.rs
fn bubble_sort(arr: &mut [i32]) {
    let n = arr.len();

    for i in 0..n {
        let mut swapped = false;

        for j in 0..(n - i - 1) {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
                swapped = true;
            }
        }

        // If no elements were swapped in this pass, array is sorted
        if !swapped {
            break;
        }
    }
}

fn main() {
    let mut arr = [64, 34, 25, 12, 22, 11, 90];
    println!("Before sorting: {:?}", arr);

    bubble_sort(&mut arr);

    println!("After sorting:  {:?}", arr);
}
